File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        PriorityQueue<Integer> pq = new PriorityQueue<Integer>();
        int[] answer = new int[score.length];
        
        for(int i = 0; i < score.length; i++){
            if(pq.size() < k){
                // 만약 pq의 사이즈가 k보다 작으면 그냥 넣으면 됨.
                pq.add(score[i]);
                System.out.println(pq.toString());
            }
            else{
                int a = pq.peek();
                if(a < score[i]){
                    pq.remove();
                    pq.add(score[i]);
                }
            }
            answer[i] = pq.peek();
        }

        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
import heapq

def solution(k, score):
    answer = []
    hq = []
    
    for i in range(len(score)):
        if len(answer) < k:
            heapq.heappush(hq, score[i])
        else:
            if hq[0] < score[i]:
                heapq.heappop(hq)
                heapq.heappush(hq, score[i])
        answer.append(hq[0])
        
    return answer